ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/SerialTreeParser.java

```diff
@@ -49,7 +49,8 @@
  * @since 2.0.0
  */
 public class SerialTreeParser extends TrialTreeParser {
-    private static final Set<String> RESTRICTED_FUNCTION_NAMES = Set.of("main", "init");
+    private static final Set<String> RESTRICTED_FUNCTION_NAMES = Set.of("main", "init", "__java_recall",
+            "__java_memorize", "__recall_any", "__recall_any_error", "__memorize", "__stmts", "__run");
```

Comment:
Shall we move this to a common place. Since similar restricted function names are listed in two places. In future, we might miss one of it when doing changes in the future.

Reply:
Updated in 8c406988f7076b9648314ab3530620ad55f0b384

Reply:
As discussed, let's introduce a simple fix by moving that to a constant class.